engine/rpcserver.go

```diff
@@ -1043,14 +1128,15 @@ func (s *RPCServer) SubmitOrder(_ context.Context, r *gctrpc.SubmitOrderRequest)
 	}
 
 	submission := &order.Submit{
-		Pair:      p,
-		Side:      order.Side(r.Side),
-		Type:      order.Type(r.OrderType),
-		Amount:    r.Amount,
-		Price:     r.Price,
-		ClientID:  r.ClientId,
-		Exchange:  r.Exchange,
-		AssetType: a,
+		Pair:          p,
+		Side:          order.Side(r.Side),
+		Type:          order.Type(r.OrderType),
+		Amount:        r.Amount,
+		Price:         r.Price,
+		ClientID:      r.ClientId,
+		ClientOrderID: r.ClientId,
```

Comment:
I am actually unclear when ClientID is used versus ClientOrderID. I noticed that GRPC is mixing this even further because it just uses ClientID then, even if ClientOrderID is previously used.

Reply:
Yeah this looks like a duplicated field, we use ClientID throughout the framework for client submitted order ID's. The only other time we use ClientID's for exchanges are ones which require a client ID as part of their API key credentials but that isn't related to orders